src/emulator/functionsEmulator.ts

```diff
@@ -1077,16 +1078,23 @@ export class FunctionsEmulator implements EmulatorInstance {
     trigger?: EmulatedTriggerDefinition
   ): Promise<Record<string, string>> {
     let secretEnvs: Record<string, string> = {};
-
+    // Extension secret overrides live in {instanceID}.secret.local, CF3 secrets live in .secret.local
+    const secretsFile = backend.extensionInstanceId
+      ? `${backend.extensionInstanceId}${LOCAL_SECRETS_FILE}`
+      : LOCAL_SECRETS_FILE;
+    const secretDirectory = backend.extensionInstanceId
+      ? path.join(this.args.projectDir, ENV_DIRECTORY)
+      : backend.functionsDir;
+    const secretPath = path.join(secretDirectory, secretsFile);
```

Comment:
seems like a good candidate for a function, then you can unit test it

Reply:
Good call - separated out and unit tested